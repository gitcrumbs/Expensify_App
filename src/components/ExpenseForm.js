import React from 'react'



export default class ExpenseForm extends React.Component{


    //Setup note change
    //setup onChange and value for the text area
  constructor(props){
    super(props);
    this.state={
        amount: props.expense?props.expense.amount:'',
        description : props.expense?props.expense.description:'',
        noteChange : props.expense?props.expense.note:'',
        createdAt : props.expense?props.expense.createdAt:'' ,
        calendarfocused: false,
        fieldError:''
       
    };
    this.onFormSubmit= this.onFormSubmit.bind(this);
  }
  


     onDescriptionChange = (e)=>{
        const description = e.target.value;
        this.setState(()=>({description}))
    }


    onnoteChange = (e)=>{
        const noteChange = e.target.value;
        this.setState(()=>({noteChange}))
    }

    onAmountChange=(e)=>{
        const amount = e.target.value;
        if(!amount||amount.match(/^\d{1,}(\.\d{0,2})?$/)){
           this.setState(()=>({amount}))
        }
       
       
    }

    onDateChange=(e)=>{
      const createDateval = e.target.value;    
      console.log("Recevied Date from Calendar",createDateval)   
     if(createDateval){
        this.setState(()=>({
            createdAt:createDateval
        }));
      }     
    }

    onFormSubmit(e){
        e.preventDefault();
       
        if(!this.state.description||!this.state.amount){           
                this.setState(()=>({
                    fieldError: 'Please Provide Description and Amount'
                })); 
        }else{
            this.setState(()=>({
                fieldError: ''
            }));
            
           this.props.onSubmit({
            description: this.state.description,
            amount : parseFloat(this.state.amount),
            createdAt : this.state.createdAt,
            note : this.state.noteChange


           });

           
        }
    }

    render(){

        return(
            <div>
                {this.state.fieldError&&<p>{this.state.fieldError}</p>}
                <form onSubmit={this.onFormSubmit}>
                    <input type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />  
                     
                    <input type="text"
                    placeholder="Amount"
                    autoFocus   
                    value={this.state.amount}                             
                    onChange={this.onAmountChange}
                  
                    />  
                    <input type="date" id="createdDate" 
                    name="createdDate"
                    value={this.state.createdAt}
                    onChange={this.onDateChange}
                    />
                    
                      <textarea
                        placeholder="Add a note for your expense Options"
                        value={this.state.noteChange}      
                        onChange={this.onnoteChange}
                      />
                    <button>Save Expense</button>
                </form>
            </div>
        );


    }
    
}