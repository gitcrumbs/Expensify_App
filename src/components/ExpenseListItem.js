import React from 'react';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral'

const ExpenseListItem = ({id,description,amount,createdAt,note})=>(

    <div>

        <Link className="expense_link" to={`/${id}`}>
            <h3>{description}</h3>            
        </Link>
        <div className="card"> 
            
                <p><b> Details :</b>{numeral(amount).format('$0,0.00')} - {moment(createdAt).format('MMMM Do, YYYY')}</p>
                <p><b>Summary :</b>{note}</p>
                  
        </div>
        
    </div>
)

export default ExpenseListItem;