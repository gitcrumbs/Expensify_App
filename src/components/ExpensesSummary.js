import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpenseSummary=({expenseCount,expensesTotal})=>{

    const expenseWord = expenseCount===1 ? 'expense':'expenses'
    const formattedExpensesTotal = numeral(expensesTotal).format('$0,0.00')
    return(
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling to {formattedExpensesTotal}</h1>
        </div>
    )
}

const mapstatetoProps = (state) =>{

    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expenseCount:visibleExpenses.length,
        expensesTotal : selectExpensesTotal(state.expenses)
    }
};

export default connect(mapstatetoProps)(ExpenseSummary);