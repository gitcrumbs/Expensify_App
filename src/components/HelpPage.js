import React from 'react';
const HelpPage = ()=>(
    <div>This is HelpPage Page </div>
    )
    
    export default HelpPage