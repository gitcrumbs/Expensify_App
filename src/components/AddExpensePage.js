import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux'
import { AddExpense } from '../actions/expenses';

const AddExpensePage = (props)=>(
    <div>
        This is from My Create Expense Page
        <ExpenseForm
            onSubmit={(expense)=>{              
                props.dispatch(AddExpense(expense))
                alert("Added Successfully for "+expense.description);
                props.history.push('/')
            }}
        />  
    </div>
)


export default connect()(AddExpensePage);
