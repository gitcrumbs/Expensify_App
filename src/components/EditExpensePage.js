import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpensebyId,removeExpensebyId} from '../actions/expenses'


const EditExpensePage = (props)=>{   
    
    return(
        <div>
            <ExpenseForm
                expense={props.expense}
                onSubmit={(expense)=>{                    
                    props.dispatch(editExpensebyId(props.expense.id,expense));
                    props.history.push('/')
                }}  
            />

            <button onClick={()=>{
                console.log("Button Click")
                props.dispatch(removeExpensebyId({id:props.expense.id}));
                props.history.push('/')
    
            }}>Remove</button>
        </div>
    );
   
};

const mapStatetoProps=(state,props)=>{
    return{

        expense : state.expenses.find((expense)=>expense.id===props.match.params.id)
    };
};

export default connect(mapStatetoProps)(EditExpensePage);