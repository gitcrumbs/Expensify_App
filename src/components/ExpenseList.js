import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'



const ExpenseList = (props) =>(
    <div>
        
            <h1>Expense List</h1>
            {props.text}
            <br/>
            {props.expenses.map((expense)=>{
                return(
                    <ExpenseListItem key={expense.id} {...expense}>
                        <p>props.note</p>
                    </ExpenseListItem>
                )
            })}
        
    </div>
);


const mapStateToprops = (state)=>{
    return{
        expenses : selectExpenses(state.expenses,state.filters)
    };
}
 
export default connect(mapStateToprops)(ExpenseList);