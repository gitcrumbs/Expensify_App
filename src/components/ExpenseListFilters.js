import React from 'react'
import ExpenseList from './ExpenseList';
import {connect} from 'react-redux'
import { setTextFilter, sortByDate,sortByAmount } from '../actions/filters';

const ExpenseListFilters = (props)=>(
    <div className="content-container">
        <div className="input-group">
        <div className="input-group__item">
        <input type="text" defaultValue={props.filters.text} onChange={(e)=>{
            props.dispatch(setTextFilter(e.target.value))
            console.log(e.target.value);
        }}/>
        </div>

        <div className="input-group__item">
                <select 
                
                onChange={(e)=>{
                    if(e.target.value==='date'){
                        props.dispatch(sortByDate())
                    }else if(e.target.value==='amount'){
                        props.dispatch(sortByAmount())
                    }
                
                }}
                >
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
        </div>
        
        </div>
        

      
        
    </div>
);

const mapStateToprops = (state)=>{
    return{
        filters : state.filters
    }
}

export default connect(mapStateToprops)(ExpenseListFilters)