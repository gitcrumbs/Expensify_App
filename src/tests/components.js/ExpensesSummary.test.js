import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import {shallow,configure} from 'enzyme';
import { ExpenseSummary } from "../../components/ExpensesSummary";


configure({adapter: new Adapter()});
test('Should correctly render ExpensesSummary with 1 expense',()=>{
    const wrapper = shallow(
        <ExpenseSummary
            expenseCount={1} expensesTotal={235}
        />
    );
    expect(wrapper).toMatchSnapshot();
});


test('Should correctly render ExpensesSummary with multiple expense',()=>{
    const wrapper = shallow(
        <ExpenseSummary
            expenseCount={23} expensesTotal={134543325}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
