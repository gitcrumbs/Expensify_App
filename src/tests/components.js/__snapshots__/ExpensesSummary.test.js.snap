// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary with 1 expense 1`] = `ShallowWrapper {}`;

exports[`Should correctly render ExpensesSummary with multiple expense 1`] = `ShallowWrapper {}`;
